src/app/data-explorer/selection-expanded-dialog/selection-expanded-dialog.component.html

```diff
@@ -0,0 +1,39 @@
+<!--
+Copyright 2022-2023 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+<div mat-dialog-title class="dialog-title">
+  <h1>Selected data elements</h1>
+  <button mat-icon-button aria-label="close dialog" mat-dialog-close>
+    <mat-icon>close</mat-icon>
+  </button>
+</div>
+<mat-dialog-content>
+  <p>{{ (selectedElements$ | async)?.length }} elements selected</p>
```

Comment:
I've noticed your use of the `async` pipe which looked interesting. Seeing as it is used in multiple places though, I wanted to check if it was being continually called. I updated the `SelectionService` to print this to the console to check:

```ts
public get list$(): Observable<DataElementSearchResult[]> {
    return this.listSubject
      .asObservable()
      .pipe(tap(() => console.log('Getting selection list...')));
  }
```

And this is the result:

![image](https://user-images.githubusercontent.com/3219480/223698990-b7e2fd78-b77c-4ee0-87f7-7c442eecb7a0.png)

This number shoots up as I move the mouse around the page too. I think what is happening is the Angular change detector keeps thinking these controls are changing, which happens on DOM event handlers. This is actually a pitfall of Angular change detection, because your templates are binding to a _function_ instead of a _value_ (even a `get` property is treated as a function), so Angular keeps calling the function to see if has changed. I would have though the `async` pipe avoids this because it is actually suggested by the Angular team themselves though.

To get around this issue though, I would suggest have your components call the `SelectionService.list$` property once, subscribe to it, then set the returned value to a local field in the component and bind to that in the template. This should prevent the change detector from going crazy.

Reply:
I've just remembered something else if going down this route. Looking at your `SelectionService`, the `list$` is an observable that doesn't complete, so you will need to manually unsubscribe from it to avoid memory leaks. There are articles explaining this and a way of declaratively unsubscribing which you may find useful. Here is one:

https://medium.com/angular-in-depth/the-best-way-to-unsubscribe-rxjs-observable-in-the-angular-applications-d8f9aa42f6a0

Reply:
I have fixed both of these bugs: the results of calling asObservable are now stored by the constructor and made available via a public member without a getter. I have replaced a getter with a public member in one of the other components too. Finally I added a `takeUntil(this.unsubscribe$)` to the subscription in search-listing.

Reply:
Re-tested usage of the `list$` observable by intercepting it with `tap(() => console.log())`, it now performs much better and only when actually required 👍